velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -408,4 +408,66 @@ struct DayOfYearFunction {
     result = getDayOfYear(getDateTime(date));
   }
 };
+
+template <typename T>
+struct NextDayFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Date>* /*startDate*/,
+      const arg_type<Varchar>* dayOfWeek) {
+    weekDay_ = getDayOfWeekFromString(*dayOfWeek);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Date>& result,
+      const arg_type<Date>& startDate,
+      const arg_type<Varchar>& dayOfWeek) {
+    VELOX_USER_CHECK(
+        !weekDay_.has_value(), "Illegal input for day of week: {}", dayOfWeek)
```

Comment:
Is this the right check? Does this function expect only constant values for dayOfWeek? Is it not possible for initialize to be called with null dayOfWeek? Can you run Fuzzer and make sure there are no failures?

CC: @PHILO-HE

Reply:
@mbasmanova This part has been modified to align with Spark. If there is an invalid ```dayOfWeek```,  null will be returned.